### DISCUSSION TITLE
Support on cmake for gtest installed

### DISCUSSION THREAD

**Discussion by inglor** [polybar/polybar#2721] [Ideas]
Hi,

Currently the tests require GoogleTest which automatically download and compile if tests are enabled. Since some distros (including Arch Linux) provide gtest suite [1] as a package and cmake has a way to detecting that [2] [3] maybe it would be a good idea to first check if `gtest` is available and then, if not, download as fallback.

Cheers,

[1]: https://archlinux.org/packages/community/x86_64/gtest/
[2]: find_package(GTest REQUIRED CONFIG)
[3]: https://github.com/ospray/ospray/blob/master/apps/ospTestSuite/CMakeLists.txt#L9

**Comment by patrick96** [comment]
I'm a bit worried about version mismatches here. Since we target the head of the `main` branch and not any specific version, gtest distro packages may stop working as soon as we use any kind of new feature.

And with gtest now following the ["live at head"-philosophy](https://github.com/google/googletest#live-at-head) there may not even be new releases, so any package that doesn't regularly update to the newest commit may be terribly outdated.

I think using system gtest should at least be behind a cmake flag.

PRs are welcome :)
